meta-balena-common/recipes-core/systemd/disk-watchdog/disk-watchdogd.service

```diff
@@ -0,0 +1,16 @@
+[Unit]
+Description=Disk watchdog service for monitoring disk health
+After=local-fs.target
+
+[Service]
+Type=notify
+ExecStart=@DAEMON_PATH@ -v -f @WD_TEST_FILE@ -i 5000
+WatchdogSec=30s
+StartLimitInterval=2min
+StartLimitBurst=3
+StartLimitAction=reboot-force
```

Comment:
Also, consider actually logging the number of reboots so that there is some traceability of the failures.